File: Part_2/Python/is_user_in_stance/is_user_in_stance.py
"""
Implement the following function

Description:
Predicts when the user is in stance (i.e. has their foot on the ground)

Input:
data_array >> an N x M numpy ndarray of variables collected from a single user
while doing a single movement for multiple steps (i.e. 10 walking steps,
10 running steps, etc).

Output:
is_user_in_stance_predictions >> an N x 1 array, where each element in the array
represents whether or not the column in data_array representing gait_state will have a value of 1
(i.e. the method should predict a value of 1 when
data_array[n,hf.variable_name_to_column_index('gait_state')] = 1 (i.e. stance) and should predict
a value of 0 when data_array[n,hf.variable_name_to_column_index('gait_state')] != 1.
"""

from copy import deepcopy
import helper_functions as hf  # pylint: disable=import-error


def is_user_in_stance(data_array):
	"""
	This is a placeholder so the code will run. Replace this with your
	implementation.
	"""
	is_user_in_stance_predictions = deepcopy(
		data_array[:, hf.variable_name_to_column_index("gait_state")]
	)
	return is_user_in_stance_predictions

File: Archive/2021_Interview_Materials/Part_2/Matlab/isUserInStance/isUserInStance.m
%% Implement the following function
% Description:
% Predicts when the user is in stance (i.e. has their foot on the ground)

% Input:
% data_table >> an N x M table of variables collected from a single user
% while doing a single movement for multiple steps (i.e. 10 walking steps,
% 10 running steps, etc).

% Output:
% isUserInStancePredictions >> an N x 1 double, where each element in the array
% represents whether or not data_table.gait_state(n) will have a value of 1
% (i.e. the method should predict a value of 1 when data_table.gait_state(n) = 1 (i.e. stance)
% and should predict a value of 0 when data_table.gait_state(n) != 1.

function isUserInStancePredictions = isUserInStance(data_table)
    % This is a placeholder so the code will run. Replace this with your
    % implementation.
    isUserInStancePredictions = data_table.gait_state;
end
